rootex/core/resource_files/material_resource_file.h

```diff
@@ -17,6 +17,8 @@ class MaterialResourceFile : public ResourceFile
 	bool saveMaterialData(const JSON::json& j);
 
 	virtual const Shader* getShader() const = 0;
+	virtual Vector<Ref<Texture>> getTextures() const { return Vector<Ref<Texture>> {}; }
```

Comment:
I included a default implementation so that it does not need to be implemented by other resource files (we only need it for basic material resource which is used by sprite component for calculating the aspect ratio of the image)